crates/red_knot_python_semantic/src/types.rs

```diff
@@ -107,41 +107,38 @@ fn widen_type_for_undeclared_public_symbol<'db>(
 }
 
 #[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
-pub(crate) enum SymbolLookup {
-    /// Look up the symbol as seen from within the same module.
-    Internal,
-    /// Look up the symbol as seen from outside the module.
-    External,
+enum RequiresExplicitReExport {
+    Yes,
+    No,
 }
 
-impl SymbolLookup {
-    const fn is_external(self) -> bool {
-        matches!(self, Self::External)
+impl RequiresExplicitReExport {
+    const fn is_yes(self) -> bool {
+        matches!(self, RequiresExplicitReExport::Yes)
     }
 }
 
-/// Infer the public type of a symbol (its type as seen from outside its scope).
-fn symbol<'db>(
+fn symbol_impl<'db>(
     db: &'db dyn Db,
-    lookup: SymbolLookup,
     scope: ScopeId<'db>,
     name: &str,
+    requires_explicit_reexport: RequiresExplicitReExport,
 ) -> Symbol<'db> {
     #[salsa::tracked]
     fn symbol_by_id<'db>(
         db: &'db dyn Db,
-        lookup: SymbolLookup,
         scope: ScopeId<'db>,
         is_dunder_slots: bool,
```

Comment:
@AlexWaygood had an idea on how we could remove this argument. But maybe that's something for its own PR (or something that @AlexWaygood can PR?)

Reply:
Oh, sure. I didn't realise we had it here already.

Reply:
I thought of using a `bitflags` i.e., combining `is_dunder_slots` and `require_explicit_reexport` unless Alex's plan is to do something different like removing it?

Reply:
https://github.com/astral-sh/ruff/pull/16138